brainiak/reprsimil/brsa.py

```diff
@@ -1659,3 +2460,1350 @@ def _loglike_AR1_singpara(self, param, XTX, XTDX, XTFX, YTY_diag,
+            to share the same pseudo-SNR map with the design matrix.
+            Therefore, the pseudo-SNR map will only reflect the
+            relative contribution of design matrix to each voxel.
+            You can provide time courses such as those for head motion
+            to this parameter.
+            Note that if auto_nuisance is set to True, the first
+            n_nureg principal components of residual (excluding the response
+            to the design matrix and the user-provided nuisance regressors)
+            will be included as additional nuisance regressor after the
+            first round of fitting.
+            If auto_nuisance is set to False, the nuisance regressors supplied
+            by the users together with DC components will be used as
+            nuisance time series.
+        scan_onsets: optional, list numpy arrays, shape=[runs,] for each.
+            Each item in the list specifies the indices of X which correspond
+            to the onset of each scanning run for one participant.
+            For example, if you have two experimental runs of
+            the first participant, each with 100 TRs, and one run of the
+            second participant, with 150 TR, then scan_onsets should be
+            [ndarry([0, 100]), ndarry([150])].
+            The effect of this argument is to make the inverse matrix
+            of the temporal covariance matrix of noise block-diagonal.
+            If you do not provide the argument, the program will
+            assume all data are from the same run for each participant.
+        """
+
+        logger.info('Running Marginalized Bayesian RSA')
+        # Checking all inputs.
+        X = self._check_data_GBRSA(X)
+        design = self._check_design_GBRSA(design, X)
+        nuisance = self._check_nuisance_GBRSA(
+            copy.deepcopy(nuisance), X)
+        # The reason that we use copy of nuisance is because they
+        # may be modified inside our code.
+
+        scan_onsets = self._check_scan_onsets_GBRSA(scan_onsets, X)
+        # Run Marginalized Bayesian RSA
+        # Note that we have a change of notation here.
+        # Within _fit_RSA_marginalized, design matrix is named X
+        # and data is named Y, to reflect the
+        # generative model that data Y is generated by mixing the response
+        # X to experiment conditions and other neural activity.
+        # However, in fit(), we keep the tradition of scikit-learn that
+        # X is the input data to fit and y, a reserved name not used, is
+        # the label to map to from X.
+        assert self.SNR_bins >= 5 and self.rho_bins >= 5, \
+            'At least 5 bins are required to perform the numerical'\
+            ' integration over SNR and rho'
+        assert self.logS_range * 6 / self.SNR_bins < 0.5, \
+            'The minimum grid of log(SNR) should not be larger than 0.5.'\
+            ' Please consider increasing SNR_bins or reducing logS_range'
```

Comment:
The first is now in test_gbrsa.py. The second is hard to define for the exponential distribution as a prior. The way I am doing is chunking [0, inf) to finite bins, and choose the expected mean of the exponential distribution of each bin as the grid to integrate. So the first one is by default large than 0.